### PR TITLE
Mw/fix pipeline 1 1 6

### PR DESCRIPTION
**Pull request by wilkermichael**
Changes proposed in this PR:
- Updated cloud acceptance test terraform files to fix issues with nightly tests
- updated `main.tf` for gke, eks and aks to use the following block as the old way of setting provider version was deprecated:
```hcl
terraform {
  required_providers {
    aws = {
      version = ">= 4.0.0"
    }
  }
}
```
- Updated aks to latest supported K8s version
- Updated eks cluster to latest supported K8s version
- gke is still using 1.25, which I think is fine for now

How I've tested this PR:
- Manually trigger nightly test: https://github.com/hashicorp/consul-k8s-workflows/actions/runs/5193186962/jobs/9378714025

How I expect reviewers to test this PR:
👀 

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/test/terraform/aks/main.tf [hunk 1]
```diff
@@ -48,7 +55,7 @@ resource "azurerm_kubernetes_cluster" "default" {
   location                          = azurerm_resource_group.default[count.index].location
   resource_group_name               = azurerm_resource_group.default[count.index].name
   dns_prefix                        = "consul-k8s-${random_id.suffix[count.index].dec}"
-  kubernetes_version                = "1.24.6"
+  kubernetes_version                = "1.26"
```
Comment (curtbushko) [inline charts/consul/test/terraform/aks/main.tf:58]: Should this be set to a minor version? Or are you thinking that if we keep at 1.26 we will be ok until the next upgrade?

File: charts/consul/test/terraform/eks/main.tf [hunk 2]
```diff
@@ -61,7 +68,7 @@ module "eks" {
   kubeconfig_api_version = "client.authentication.k8s.io/v1beta1"
 
   cluster_name    = "consul-k8s-${random_id.suffix[count.index].dec}"
-  cluster_version = "1.23"
+  cluster_version = "1.26"
```
Comment (curtbushko) [inline charts/consul/test/terraform/eks/main.tf:71]: Same comment as above about the version.

**Comment by wilkermichael** [comment]
Talked to @thisisnotashwin , and he thinks it would be a good idea to spread the tests over the supported kubernetes versions. So I will address that in a follow up PR.

File: charts/consul/test/terraform/eks/main.tf (hunk 2 above)
Comment (wilkermichael) [reply charts/consul/test/terraform/eks/main.tf:71]: Talked to @thisisnotashwin I'll pin these to a patch version as well for better repro-ability

**Pull request merged**